typhos/related_display.py

```diff
@@ -0,0 +1,121 @@
+"""
+Widgets that open up typhos displays.
+"""
+from happi.client import Client
+from happi.loader import load_devices
+from pydm.utilities import establish_widget_connections, is_qt_designer
+from qtpy import QtCore, QtWidgets
+
+from .suite import TyphosSuite
+from .utils import TyphosObject, no_device_lazy_load, use_stylesheet
+
+
+class TyphosRelatedSuiteButton(TyphosObject, QtWidgets.QPushButton):
+    """
+    Button to open a typhos suite with happi-loaded devices.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._happi_names = []
+        self._happi_cfg = ''
+        self._preload = False
+        self._suite = None
+        self.clicked.connect(self.show_suite)
+
+    @QtCore.Property('QStringList')
+    def happi_names(self):
+        """
+        List of devices to include in the suite.
+        """
+        return self._happi_names
+
+    @happi_names.setter
+    def happi_names(self, happi_names):
+        self._happi_names = happi_names
+
+    @QtCore.Property(str)
+    def happi_cfg(self):
+        """
+        Happi config to use, or empty string to use environment variable.
+        """
+        return self._happi_cfg
+
+    @happi_cfg.setter
+    def happi_cfg(self, happi_cfg):
+        self._happi_cfg = happi_cfg
+
+    @QtCore.Property(bool)
+    def preload(self):
+        """
+        If True, we'll create the suite ahead of time.
+        """
+        return self._preload
+
+    @preload.setter
+    def preload(self, exec_preload):
+        self._preload = exec_preload
+        if self._preload and not is_qt_designer():
+            self.create_suite()
```

Comment:
`for i in range(10): widget.preload = True`
Not a likely scenario, but probably something to guard against

Reply:
Good catch, I'll resolve by adding a condition that checks if we've already created the suite.